src/praisonai-agents/praisonaiagents/bots/presentation.py

```
@@ -624,41 +655,70 @@ def _adapt_button(button: PresentationButton, limits: PresentationLimits) -> Pre
+    recovered — see the issue this addresses).
     """
     raw = f"select:{action_id}:{value}"
     if len(raw.encode("utf-8")) <= _MAX_CALLBACK_LEN:
         return raw
+    if store is not None:
+        ref = _callback_ref(f"select:{action_id}", value)
+        store.put(ref, value, expires_at=time.time() + CALLBACK_REF_TTL)
+        prefix = f"select:{action_id}:{CALLBACK_REF_MARKER}"
+        if len(prefix.encode("utf-8")) + len(ref) > _MAX_CALLBACK_LEN:
+            # action_id itself is long; hash it so the ref still fits the bound.
+            aid_digest = hashlib.sha1((action_id or "").encode("utf-8")).hexdigest()[:8]
```

Ruff (0.15.21)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)